### ISSUE TITLE
This plugin doesn't work on Catalina

### ISSUE THREAD

**Issue by pawsys** [sballin/alfred-search-notes-app#12]

**Comment by sballin** [comment]
Are you using the latest version, 1.4.3? If it's still not working please post the text in the [debug log](https://www.alfredapp.com/help/workflows/advanced/debugger/) after you try to find and open a note. Take care to remove any private note contents if they show up there.

**Comment by pawsys** [comment]
Does it work for you using Alfred 4 / Catalina? (yes I am using the newest
version)

**Comment by sballin** [comment]
It works for me in Alfred 3 and Catalina. If you post the info from the debug log I may be able to help.

**Comment by pawsys** [comment]
How to run a debug log? I am typing the shortcut for the notes app and no
notes are showing up.

**Comment by sballin** [comment]
Follow the instructions here to open the debug log: https://www.alfredapp.com/help/workflows/advanced/debugger/

Then try to use the workflow and see the output that appears in the debug log.

**Comment by tiwood** [comment]
It seems Apple Script support in the Notes.app is broken in Catalina if iCloud Sync is enabled.
The following thread is from another application but the issue seems to be the same:
https://hookproductivity.com/help2/integration/hook-and-macos-10-15-catalina-notes-app/

This is the error I'm getting right now:
```
[8:11:25.813 PM] ERROR: Search Notes.app[Run NSAppleScript] {
    NSAppleScriptErrorAppName = Notes;
    NSAppleScriptErrorBriefMessage = "AppleEvent handler failed.";
    NSAppleScriptErrorMessage = "Notes got an error: AppleEvent handler failed.";
    NSAppleScriptErrorNumber = "-10000";
    NSAppleScriptErrorRange = "NSRange: {67, 20}";
```

**Comment by sballin** [comment]
Thanks for the helpful info! Weirdly, I can't reproduce this. I was testing with "on my mac" notes on Catalina 10.15 + Alfred 3 before, but I upgraded to 10.15.1 and added my iCloud notes, and it continued to work. In Script Editor.app, can you try the following?

```
tell application "Notes" to return every note in default account
```

then after getting the list of `x-coredata://...` urls, copy one of them and try

```
tell application "Notes" to show note id "x-coredata://..." in default account
```

You can also play with removing `in default account`.

**Comment by sballin** [comment]
One more question, your error happens after you successfully get the list of notes then try to open one, right?

I looked into whether it might be a permissions thing. In System Preferences -> Security & Privacy -> Privacy tab, under "Automation" I gave permission to Notes, which appears with a checkbox under Alfred 3. When I removed that permission, I got a different error, number -1743 with message "Not authorized to send Apple events to Notes."

**Comment by tiwood** [comment]
The error happens in the `Show note or folder with given ID` RunNSAppleScript action. The Notes listing works fine for me. Additionally I've checked the permissions and they look ok to me:

<img width="197" alt="screenshot_2019-11-13_20-46-40" src="https://user-images.githubusercontent.com/6249819/68798440-b8774600-0656-11ea-8334-19dc011a6c06.png">

-----
Update: I've just added `in default account` to the AppleScript and now it works!

```
on alfred_script(q)
	tell application "Notes"
		if q contains "Note"
			show note id q in default account
		else
			show folder id q in default account
		end if
	end tell
end alfred_script
```

I'm only storing notes in my default account, so this is a good workaround for me.
